sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -183,15 +187,24 @@ export class AppConfigurationClient {
       async (updatedOptions) => {
         const keyValue = serializeAsConfigurationSettingParam(configurationSetting);
         logger.info("[addConfigurationSetting] Creating a key value pair");
-        const originalResponse = await this.client.putKeyValue(configurationSetting.key, {
-          ifNoneMatch: "*",
-          label: configurationSetting.label,
-          entity: keyValue,
-          ...updatedOptions,
-        });
-        const response = transformKeyValueResponse(originalResponse);
-        assertResponse(response);
-        return response;
+        try {
+          const originalResponse = await this.client.putKeyValue(configurationSetting.key, {
+            ifNoneMatch: "*",
+            label: configurationSetting.label,
+            entity: keyValue,
+            ...updatedOptions,
+          });
+          const response = transformKeyValueResponse(originalResponse);
+          assertResponse(response);
+          return response;
+        } catch (error) {
+          const err = error as RestError;
+          if (err.statusCode === 412) {
+            err.message = `Status 412: Setting was already present`;
```

Comment:
Can you also add a comment stating that the service doesn't provide an error message and that we are doing this in sync with other languages?